lib/src/components/NxCombobox/__tests__/NxCombobox.test.tsx

```diff
@@ -0,0 +1,522 @@
+        inputElement = getByRole('combobox'),
+        dropdownElement = getByRole('listbox', { hidden: true });
+
+    expect(dropdownElement).toHaveAttribute('id');
+    expect(inputElement).toHaveAttribute('aria-controls', dropdownElement.id);
+  });
+
+  it('sets the alert role on the alert dropdown when it is in loading, error, or empty states', function() {
+    const { getAllByRole, rerender } = quickRender({ value: 'f' }),
+        alertDropdownElement = getAllByRole('alert', { hidden: true })[1];
+
+    expect(alertDropdownElement).toBeInTheDocument();
+
+    rerender(<NxCombobox { ...minimalProps } loading={true} />);
+    expect(alertDropdownElement).toBeInTheDocument();
+
+    rerender(<NxCombobox { ...minimalProps } loadError={'boo'} />);
+    expect(alertDropdownElement).toBeInTheDocument();
+  });
+
+  it('sets aria-live on the alert dropdown to "polite"', function() {
+    expect(quickRender().getAllByRole('alert', { hidden: true })[1]).toHaveAttribute('aria-live', 'polite');
+  });
+
+  it('sets aria-busy on the alert dropdown if loading is true', function() {
+    const { getAllByRole, rerender } = quickRender(),
+        alertDropdownElement = getAllByRole('alert', { hidden: true })[1];
+
+    expect(alertDropdownElement).toHaveAttribute('aria-busy', 'false');
+
+    rerender(<NxCombobox { ...minimalProps } loading={true} />);
+    expect(alertDropdownElement).toHaveAttribute('aria-busy', 'true');
+  });
+
+  it('renders error and textcontent when the `loadError` prop is set', function() {
+    const alertDropdownElement = quickRender({ loadError: 'err' }).getAllByRole('alert', { hidden: true })[1];
+
+    expect(alertDropdownElement).toBeInTheDocument();
+    expect(alertDropdownElement).toHaveTextContent('err');
+  });
+
+  it('fires onSearch with the search text when the retry button is clicked', async function() {
+    const onSearch = jest.fn(),
+        { getByRole} = quickRender({ loadError: 'err', onSearch }),
+        inputElement = getByRole('combobox');
+
+    expect(onSearch).not.toHaveBeenCalled();
+    await userEvent.type(inputElement, 'f');
+    const retryBtn = getByRole('button', { name: /retry/i });
+    await userEvent.click(retryBtn);
+    expect(onSearch).toHaveBeenCalledWith('f');
+  });
+
+  it('fires onChange when the button with role option is clicked', async function() {
+    const onChange = jest.fn(),
+        { getByRole, getAllByRole } = quickRender({
+          matches: [{ id: '1', displayName: 'Foo' }, { id: '2', displayName: 'Boo' }], onChange }),
+        inputElement = getByRole('combobox');
+
+    await userEvent.type(inputElement, 'o');
+    const optionBtns = getAllByRole('option');
+    await userEvent.click(optionBtns[1]);
+    expect(onChange).toHaveBeenCalledWith('Boo');
+  });
+
+  it('renders empty message if there are no results', function() {
+    const emptyMessage = quickRender().getAllByRole('alert', { hidden: true })[1];
```

Comment:
Since the alert has attribute `aria-hidden=true`, it is needed in order to query it with `getAllByRole`.

Reply:
> Since the alert has attribute aria-hidden=true, it is needed here in order to query it with getAllByRole.

This is another one where I think focusing the input would help.  The goal of this test is to assert that the user sees the empty message.  If we're asserting that the message is in the DOM, but might be hidden, that isn't really accomplishing the goal.